### PR TITLE
feat: Angles of incidence in neural calibrator

### PR DESCRIPTION
**Pull request by gagnonlg**
As stated in #2493, angles of incidence are more useful than global frame direction for calibration purposes, so in the PR we add feed these to the neural calibrator instead.

depends on #2493

### CODE REVIEW

File: Examples/Framework/ML/src/NeuralCalibrator.cpp [hunk 1]
```diff
@@ -107,6 +108,17 @@ void ActsExamples::NeuralCalibrator::calibrate(
         Acts::ActsSquareMatrix<Acts::eBoundSize> fcov =
             E * measurement.covariance() * E.transpose();
 
+        Acts::Vector3 dir = Acts::makeDirectionFromPhiEta(
+            fpar[Acts::eBoundPhi], fpar[Acts::eBoundTheta]);
+        Acts::Vector3 gpos =
+            surf.localToGlobal(gctx, fpar.segment<2>(Acts::eBoundLoc0), dir);
+        Acts::RotationMatrix3 rot =
```
Comment (felix-russo) [inline Examples/Framework/ML/src/NeuralCalibrator.cpp:115]:
```suggestion
// Rotation matrix. When applied to global coordinates, they are rotated into the local reference frame of the surface. Note that this such a rotation can be found by inverting a matrix whose columns correspond to the coordinate axes of the local coordinate system.
        Acts::RotationMatrix3 rot =
```

Better double check if I got the explanation right

File: Examples/Framework/ML/src/NeuralCalibrator.cpp [hunk 2]
```diff
@@ -96,8 +97,8 @@ void ActsExamples::NeuralCalibrator::calibrate(
 
   input[iInput++] = idxSourceLink.geometryId().volume();
   input[iInput++] = idxSourceLink.geometryId().layer();
-  input[iInput++] = trackState.parameters()[Acts::eBoundPhi];
-  input[iInput++] = trackState.parameters()[Acts::eBoundTheta];
+
+  const Acts::Surface& surf = trackState.referenceSurface();
```
Comment (felix-russo) [inline Examples/Framework/ML/src/NeuralCalibrator.cpp:101]:
```suggestion
  const Acts::Surface& referenceSurface = trackState.referenceSurface();
```
nitpick: makes the code easier to read (at least for me)

**Review by felix-russo — COMMENTED** [review]
Looks good, just a few minor comments

File: Examples/Framework/ML/src/NeuralCalibrator.cpp [hunk 3]
```diff
@@ -107,6 +108,17 @@ void ActsExamples::NeuralCalibrator::calibrate(
         Acts::ActsSquareMatrix<Acts::eBoundSize> fcov =
             E * measurement.covariance() * E.transpose();
 
+        Acts::Vector3 dir = Acts::makeDirectionFromPhiEta(
```
[the remark this answers is not in the document]
Comment (gagnonlg) [reply Examples/Framework/ML/src/NeuralCalibrator.cpp:111]: Good catch

File: Examples/Framework/ML/src/NeuralCalibrator.cpp [hunk 4]
```diff
@@ -107,6 +108,17 @@ void ActsExamples::NeuralCalibrator::calibrate(
...
+            fpar[Acts::eBoundPhi], fpar[Acts::eBoundTheta]);
+        Acts::Vector3 gpos =
```
[the remark this answers is not in the document]
Comment (gagnonlg) [reply Examples/Framework/ML/src/NeuralCalibrator.cpp:113]: Changed

File: Examples/Framework/ML/src/NeuralCalibrator.cpp (hunk 1 above)
Comment (gagnonlg) [reply Examples/Framework/ML/src/NeuralCalibrator.cpp:115]: Sounds about right

File: Examples/Framework/ML/src/NeuralCalibrator.cpp (hunk 2 above)
Comment (gagnonlg) [reply Examples/Framework/ML/src/NeuralCalibrator.cpp:101]: Done

**Comment by gagnonlg** [comment]
@felix-russo thanks for your review! The comments have been implemented.

**Review by felix-russo — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303779) |
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303283) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303282) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303281) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303280) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303279) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303278) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303277) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303276) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303275) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303274) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303273) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303272) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303271) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303269) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33303268) |

**Pull request merged**